analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LossOfFractionInDivision.CSharp10.cs

```diff
@@ -1,10 +1,63 @@
-﻿public struct S
+﻿using System;
+using System.Threading.Tasks;
+
+public struct S
 {
     public decimal Property { get; set; }
 
     public void M()
     {
-        (Property, var _) = (3 / 2, // FN
-                             3 / 2);
+        (Property, var _) = (3 / 2, 3 / 2);    // Noncompliant
+
+        (decimal x, var y) = (1 / 3, 2);      // Noncompliant
+        (decimal xx, var (yy, zz)) = (1 / 3, (2, 1 / 3)); // Noncompliant
+        (decimal xxx, var (yyy, zzz)) = (1, (2, 1 / 3));
+        (decimal xxxx, var (yyyy, zzzz)) = (1, (2, 1 / 3)); // FN
+
+
+        var (a, b) = (1 / 3, 2);
+        var (a1, (b1, c1)) = (1, (1 / 3, 2));
+        var d = (1 / 3, 2);
+        (int, int) d2 = (1 / 3, 2); 
+        (decimal, decimal) d3 = (1 / 3, 2); // Noncompliant
+        (int, (decimal, (int, decimal))) nested = (3, (1 / 3, (0, 1 / 3))); // Noncompliant [issue1, issue2]
```

Comment:
here I think it would make sense to add message and position, as well (to ensure it's correct for nested tuples)